extensions/notebook/src/dialog/remoteBookDialog.ts

```diff
@@ -0,0 +1,249 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as loc from '../common/localizedConstants';
+import { RemoteBookController, IReleases } from '../book/remoteBookController';
+
+function getRemoteLocationCategory(name: string): azdata.CategoryValue {
+	if (name === 'GitHub') {
+		return { name: name, displayName: loc.onGitHub };
+	}
+	return { name: name, displayName: loc.onSharedFile };
+}
+
+export class RemoteBookDialogModel {
+	private _remoteTypes: azdata.CategoryValue[];
+	private _controller: RemoteBookController;
+
+	constructor(
+	) {
+		this._controller = new RemoteBookController();
+	}
+
+	public get remoteLocationCategories(): azdata.CategoryValue[] {
+		if (!this._remoteTypes) {
+			this._remoteTypes = [getRemoteLocationCategory('GitHub'), getRemoteLocationCategory('Shared File')];
+		}
+		return this._remoteTypes;
+	}
+
+	public async getReleases(url: string): Promise<IReleases[]> {
+		let remotePath: URL;
+		try {
+			url = 'https://api.github.com/'.concat(url);
+			remotePath = new URL(url);
+			return this._controller.getReleases(remotePath);
+		}
+		catch (error) {
+			throw error;
+		}
+	}
+
+	public async downloadLocalCopy(url: URL, remoteLocation: string, release?: IReleases): Promise<string[]> {
+		if (release) {
```

Comment:
You can shorten this function to `return await this._controller.setRemoteBook(url, remoteLocation, release?.zipURL, release?.tarURL);` which will do the same thing. 


`